image/descriptor.go

```diff
@@ -23,25 +23,22 @@ import (
 	"strings"
 
 	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pkg/errors"
 )
 
-type descriptor struct {
-	MediaType string `json:"mediaType"`
-	Digest    string `json:"digest"`
-	Size      int64  `json:"size"`
-}
+type descriptor v1.Descriptor
```

Comment:
If not, two descriptor cannot be assigned. （Type conversion is not ok. ）

> descriptor.go:67: cannot convert &index.Manifests[i].Descriptor (type *v1.Descriptor) to type *descriptor

If you want to use the original definition as you say, you can only assign each structure separately in the original way.